src/Adapter/Category/CommandHandler/AbstractDeleteCategoryHandler.php

```diff
@@ -28,32 +28,23 @@
 
 use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\CategoryDeleteMode;
 use Product;
-use Shop;
 
 /**
  * Class AbstractDeleteCategoryHandler.
  */
 abstract class AbstractDeleteCategoryHandler
 {
     /**
-     * Handle products category after its deletion.
+     * Handle products update after category deletion.
      *
-     * @param int $parentCategoryId
-     * @param CategoryDeleteMode $mode
+     * @param array $categoryProducts products that belong to the category before deletion
+     * @param int $parentCategoryId parent category of the category to be deleted
+     * @param CategoryDeleteMode $mode deletion mode
      */
-    protected function handleProductsUpdate($parentCategoryId, CategoryDeleteMode $mode)
+    protected function handleProductsUpdate(array $categoryProducts, $parentCategoryId, CategoryDeleteMode $mode)
```

Comment:
Could you move this parameter to the end and add a default value to `[]` for avoiding BC Break ?